graylog2-web-interface/src/views/components/contexts/WidgetFocusProvider.tsx

```diff
@@ -15,27 +15,63 @@
  * <http://www.mongodb.com/licensing/server-side-public-license>.
  */
 import * as React from 'react';
-import { useState, useEffect } from 'react';
+import { useState, useEffect, useCallback } from 'react';
+import { useLocation, useHistory } from 'react-router-dom';
 import PropTypes from 'prop-types';
+import URI from 'urijs';
 
 import { useStore } from 'stores/connect';
 import { WidgetStore } from 'views/stores/WidgetStore';
 import WidgetFocusContext from 'views/components/contexts/WidgetFocusContext';
 
+const _syncWithQuery = (query: string, focusedWidget: string) => {
+  const baseUri = new URI(query)
+    .removeSearch('focused');
+
+  if (focusedWidget) {
+    return baseUri.setSearch('focused', focusedWidget).toString();
+  }
+
+  return baseUri.toString();
+};
+
+const getFocusedWidgetFromSearch = (search: string) => {
+  const uri = new URI(search);
+  const { focused } = uri.search(true);
+
+  return focused;
+};
+
 const WidgetFocusProvider = ({ children }: { children: React.ReactNode }): React.ReactElement => {
-  const [focusedWidget, setFocusedWidget] = useState(undefined);
+  const { search, pathname } = useLocation();
```

Comment:
You can use this instead:
```suggestion
  const { focused } = useQuery();
```

`useQuery` should be imported from `routing/useQuery`.
This already does the query string parsing and returns an object containing the query parameters.